pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -375,24 +398,16 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 			}
 		}
 
-		if len(in.AppendHeaders) > 0 {
-			action.RequestHeadersToAdd = make([]*core.HeaderValueOption, 0)
-			for key, value := range in.AppendHeaders {
-				action.RequestHeadersToAdd = append(action.RequestHeadersToAdd, &core.HeaderValueOption{
-					Header: &core.HeaderValue{
-						Key:   key,
-						Value: value,
-					},
-				})
-			}
-		}
+		requestHeadersToAdd := append(translateAppendHeaders(in.AppendRequestHeaders), translateAppendHeaders(in.AppendHeaders)...)
+		sort.Stable(SortHeaderValueOption(requestHeadersToAdd))
+		out.RequestHeadersToAdd = requestHeadersToAdd
 
-		if len(in.RemoveResponseHeaders) > 0 {
-			action.ResponseHeadersToRemove = make([]string, 0)
-			for _, value := range in.RemoveResponseHeaders {
-				action.ResponseHeadersToRemove = append(action.ResponseHeadersToRemove, value)
-			}
-		}
+		responseHeadersToAdd := translateAppendHeaders(in.AppendResponseHeaders)
+		sort.Stable(SortHeaderValueOption(responseHeadersToAdd))
```

Comment:
Put the `sort.Stable` inside `translateAppendHeaders`, then you can just do (here and elsewhere):
`out.ResponseHeadersToAdd = translateAppendHeaders(in.AppendResponseHeaders)`